app/packages/core/src/components/Modal/Group/DynamicGroup/UnorderedDynamicGroup.tsx

```diff
@@ -1,19 +1,72 @@
-import React from "react";
+import { Bar } from "@fiftyone/components";
+import { AbstractLooker } from "@fiftyone/looker";
+import * as fos from "@fiftyone/state";
+import React, { useRef } from "react";
+import { useRecoilValue } from "recoil";
+import styled from "styled-components";
+import { ModalActionsRow } from "../../../Actions";
 import Sample from "../../Sample";
 import { useGroupContext } from "../GroupContextProvider";
 import { DynamicGroupCarousel } from "./carousel/DynamicGroupCarousel";
 
+const RootContainer = styled.div`
+  height: 100%;
+  width: 100%;
+`;
+
+const ElementsContainer = styled.div`
+  width: 100%;
+  height: 100%;
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+`;
+
 export const UnorderedDynamicGroup = () => {
   const { lookerRefCallback, groupByFieldValue } = useGroupContext();
+  const lookerRef = useRef<AbstractLooker>();
+
+  const isImageVisible = useRecoilValue(fos.groupMediaIsImageVisible);
+  const isCarouselVisible = useRecoilValue(fos.groupMediaIsCarouselVisible);
 
   if (!groupByFieldValue) {
     return null;
   }
 
   return (
-    <>
-      <DynamicGroupCarousel key={groupByFieldValue} />
-      <Sample lookerRefCallback={lookerRefCallback} />
-    </>
+    <RootContainer>
+      {/* weird conditional rendering of the bar because lookerControls messes up positioning of the bar in firefox in inexplicable ways */}
+      {!isImageVisible && <UnorderedDynamicGroupBar lookerRef={lookerRef} />}
+      <ElementsContainer>
+        {isImageVisible && <UnorderedDynamicGroupBar lookerRef={lookerRef} />}
+        {isCarouselVisible && <DynamicGroupCarousel key={groupByFieldValue} />}
+        {isImageVisible && (
+          <Sample
+            lookerRefCallback={lookerRefCallback}
+            lookerRef={lookerRef}
+            hideSampleBar
+          />
+        )}
+      </ElementsContainer>
+    </RootContainer>
+  );
+};
+
+const UnorderedDynamicGroupBar = ({
```

Comment:
Consider:

```typescript
interface UnorderedDynamicGroupBarProps {
  lookerRef: React.RefObject<AbstractLooker>;
}

const UnorderedDynamicGroupBar: React.FC<UnorderedDynamicGroupBarProps> = ({
  lookerRef,
}) => {
  // ...
};
```

Reply:
`React.MutableRefObject<AbstractLooker | undefined>; could be replaced with React.RefObject<AbstractLooker> because MutableRefObject is expected to be initialized when it's created.